### ISSUE TITLE
Add task list progression design features to the setup task list.

### ISSUE THREAD

**Issue by samueljseay** [woocommerce/woocommerce-admin#6774]
This is to implement the new designs proposed by @elizaan36 for the setup task list. This will be dependent on completing #6772 and #6773  first.

![image](https://user-images.githubusercontent.com/1281828/114324878-e2e96d80-9b80-11eb-8df0-fba5cdbf1052.png)

Requirements
* Any task can be expanded at any time (the tasks don't have to be completed in order)
* Only one task can be expanded at a time
* Completed tasks won't expand, they'd go the settings area or task screen where data can be edited

Expanded task copy and CTA:

If store profiler was skipped:
<img width="801" alt="Screen Shot 2021-05-03 at 3 49 59 PM" src="https://user-images.githubusercontent.com/63922/116925725-ca67f280-ac16-11eb-86b1-62a754a46a3e.png">

Add products:
<img width="801" alt="Screen Shot 2021-05-03 at 3 50 12 PM" src="https://user-images.githubusercontent.com/63922/116925722-c9cf5c00-ac16-11eb-8c6f-44009b45376a.png">

WCPay installed:
<img width="666" alt="Screen Shot 2021-05-18 at 4 19 30 PM" src="https://user-images.githubusercontent.com/5121465/118725103-18731d80-b7f5-11eb-8007-e98d640558c4.png">

WCPay not installed:
<img width="661" alt="Screen Shot 2021-05-18 at 4 22 51 PM" src="https://user-images.githubusercontent.com/5121465/118725261-5112f700-b7f5-11eb-9979-73055dfc40a3.png">

Shipping:
<img width="813" alt="Screen Shot 2021-05-03 at 3 51 02 PM" src="https://user-images.githubusercontent.com/63922/116925716-c936c580-ac16-11eb-94a9-24aad4e464e6.png">

No WCShip, No location set:
<img width="662" alt="Screen Shot 2021-05-18 at 4 24 11 PM" src="https://user-images.githubusercontent.com/5121465/118725433-90d9de80-b7f5-11eb-819c-1a58c3b09fd1.png">

WCShip installed:
<img width="801" alt="Screen Shot 2021-05-03 at 3 50 39 PM" src="https://user-images.githubusercontent.com/63922/116925718-c9cf5c00-ac16-11eb-9755-51b852a9fea1.png">

Personalize your store:
<img width="674" alt="Screen Shot 2021-05-18 at 4 19 47 PM" src="https://user-images.githubusercontent.com/5121465/118724990-f4afd780-b7f4-11eb-9e8b-24b966e5e06b.png">

Multiple extensions added:
<img width="660" alt="Screen Shot 2021-05-18 at 4 26 17 PM" src="https://user-images.githubusercontent.com/5121465/118725623-cf6f9900-b7f5-11eb-9c81-4335eac29307.png">

One extension added:
<img width="663" alt="Screen Shot 2021-05-18 at 4 26 58 PM" src="https://user-images.githubusercontent.com/5121465/118725689-e2826900-b7f5-11eb-8c47-aceb492f6a24.png">

**Comment by jeffstieler** [comment]
Do we have the copy for the expanded view of the existing tasks available anywhere? @samueljseay @elizaan36

**Comment by elizaan36** [comment]
yep, I can share the designs/copy for all steps in Slack.

**Comment by louwie17** [comment]
@jeffstieler some of the discussion ran over into another ticket (it took me a little while to find), but here is an gif of some of it: https://github.com/woocommerce/woocommerce-admin/issues/6639#issuecomment-818333492

**Comment by jeffstieler** [comment]
Thanks @elizaan36 @louwie17 - I added screenshots of the copy and CTA text to the issue.

**Comment by jeffstieler** [comment]
Do we have expanded copy and a CTA for the "add paid extensions" task?

<img width="528" alt="Screen Shot 2021-05-06 at 9 55 00 AM" src="https://user-images.githubusercontent.com/63922/117310347-6f194880-ae40-11eb-9d23-968d4e22a646.png">

In the event there is only one to buy and install it is listed by name:

<img width="521" alt="Screen Shot 2021-05-06 at 9 54 27 AM" src="https://user-images.githubusercontent.com/63922/117310392-7a6c7400-ae40-11eb-848d-f433f977fb15.png">

**Comment by elizaan36** [comment]
Ideally we could replace the modal for this flow, since the extended task can contain the explicit CTA to Purchase & install.
<img width="754" alt="image" src="https://user-images.githubusercontent.com/5121465/117329564-6170bc80-ae5a-11eb-89a0-2d5da3a1e820.png">

Here are the two options for copy - let me know what you think.
Multiple extensions:
<img width="697" alt="image" src="https://user-images.githubusercontent.com/5121465/117330167-07242b80-ae5b-11eb-8890-ddcae6a896f4.png">

Single extension (Bundles example):
<img width="706" alt="image" src="https://user-images.githubusercontent.com/5121465/117329861-b1e81a00-ae5a-11eb-9592-b96a070ae847.png">

Could we use the same copy that exists for the modal for individual paid extensions?

**Comment by jeffstieler** [comment]
> I'll have fallback behavior for cases where there isn't expanded text or a CTA specified.

On second thought, I don't think I'll be handling these cases - I think all items within the same list need to have the "expand, then click the CTA" behavior otherwise it might be jarring/confusing.

> Could we use the same copy that exists for the modal for individual paid extensions?

Yes, that looks to be possible - just porting that functionality from the modal component to the task list area. In the case where multiple paid extensions are needing purchase, should we display something similar (or identical) to the existing modal?

**Comment by jeffstieler** [comment]
Looks like we need copy for the normal (non WC-Pay) payment method task titled "Choose payment methods".

**Comment by elizaan36** [comment]
> In the case where multiple paid extensions are needing purchase, should we display something similar (or identical) to the existing modal?

I shortened the copy in the screenshot I shared above. Can we just use that rather than presenting the modal?

> Looks like we need copy for the normal (non WC-Pay) payment method task titled "Choose payment methods".

That's in the Figma - here's what I've got:
<img width="704" alt="image" src="https://user-images.githubusercontent.com/5121465/117350385-1c588480-ae72-11eb-848d-98e630846f86.png">

> What should the copy be for the tax rates task if the store isn't eligible for WooCommerce Services?

Not sure on that one. Is there anything in the code that gives you an idea of what that might be?

**Comment by elizaan36** [comment]
For the WooCommerce Services tax task we wouldn't need the separate screen, but there's a stepper when WC Services isn't involved.
<img width="970" alt="image" src="https://user-images.githubusercontent.com/5121465/117350902-b6b8c800-ae72-11eb-942e-6976bfd661a5.png">

**Comment by elizaan36** [comment]
Without the store location, something like this for the task could work:
<img width="698" alt="image" src="https://user-images.githubusercontent.com/5121465/117351249-19aa5f00-ae73-11eb-9fe5-d3968fac86ea.png">
But if store location is already set, it could be one click without going to the stepper:
<img width="693" alt="image" src="https://user-images.githubusercontent.com/5121465/117351653-92112000-ae73-11eb-8363-0dee8af56374.png">
(It would be kind of annoying to click into the extra screen only to be directed to settings😆)

**Comment by jeffstieler** [comment]
Blocked by https://github.com/woocommerce/woocommerce-admin/issues/6949

**Comment by pmcpinto** [comment]
@jeffstieler @elizaan36 just some feedback about the copy:

- WooCommerce Payments task: we need to include the legal message: By setting up, you are agreeing to the [Terms of Service](https://wordpress.com/tos/)
- Payments task: I think that naming WCPay, Stripe or PayPal in the description makes it difficult to maintain. We're progressively expanding WCPay to new countries and we have other partners in EU, LATAM, APAC. 
- Personalize store task: considering we have a theme called Storefront, perhaps using "your storefront" in the description can be confusing. 

Lmk what you think. Thanks

**Comment by elizaan36** [comment]
Just updated the copy based on this feedback and added all the screenshots from different use cases mentioned the thread into the original comment!
